### DISCUSSION TITLE
Adding fonts after compiler initialization

### DISCUSSION THREAD

**Discussion by Kvnyu** [Myriad-Dreamin/typst.ts#412] [Q&A] [answered]
Hi,

Thanks for this library! I'm wondering if it is possible to add fonts after the compiler has been initialised ? I've tried importing via `.mapShadow` however this does not seem to work. 

Thanks!

**Comment by Myriad-Dreamin** [comment, marked as answer]
Hi, typst.ts doesn't support dynamic font loading, which is limited by typst compiler itself.

PS: There will be some resource type rfc for dynamic font loading, but the rfc is still in draft.
See [Discord Topic: Resource type](https://discord.com/channels/1054443721975922748/1153346652535537740)

> A name could be a fourth kind, for instance for fonts
> #set text(font: "A name")
> #set text(font: resource.path("Blabla.ttf")) 
> This means that the default would need to be adaptable though
> Because for fonts, the default should be name, not path
> I guess for all resource that can have a name, it should be the default
> Citation styles, too

**Comment by Kvnyu** [comment]
> Hi, typst.ts doesn't support dynamic font loading, which is limited by typst compiler itself.
> 
> PS: There will be some resource type rfc for dynamic font loading, but the rfc is still in draft. See [Discord Topic: Resource type](https://discord.com/channels/1054443721975922748/1153346652535537740)
> 
> > A name could be a fourth kind, for instance for fonts
> > #set text(font: "A name")
> > #set text(font: resource.path("Blabla.ttf"))
> > This means that the default would need to be adaptable though
> > Because for fonts, the default should be name, not path
> > I guess for all resource that can have a name, it should be the default
> > Citation styles, too

Ah that is what I was guessing. Thanks!

**Comment by Myriad-Dreamin** [comment]
https://github.com/Myriad-Dreamin/typst.ts/issues/634#issuecomment-3315466162
